std/format/internal/floats.d

```diff
@@ -3282,62 +3283,68 @@ private auto printFloat0(bool g, Char)(return char[] buf, FormatSpec!Char f, str
 // check no allocations
 @system unittest
 {
+    import std.format : NoOpSink;
+    auto w = NoOpSink();
+
     import core.memory;
     auto stats = GC.stats;
```

Comment:
So the GC does get called, but no allocations are performed. I wonder how we can trnasition to making the unittest `@nogc`.

Reply:
I could do

```
struct NoOpSink
{
    void put(E)(scope const E) pure @safe @nogc nothrow {}
}

auto w = NoOpSink();
```

instead. This should use the stack if I'm not completely wrong.